src/uwtools/rocoto.py

```diff
@@ -182,13 +182,60 @@ def _add_task_dependency(self, e: Element, config: dict) -> None:
         :param config: Configuration data for this element.
         """
         e = SubElement(e, STR.dependency)
+        for key, block in config.items():
+            tag, _ = self._tag_name(key)
+            if tag in [STR.taskdep, STR.datadep, STR.timedep]:
+                self._add_task_dependency_operand(e, config={key: block})
```

Comment:
Given some of my comments on how the data you're passing to these methods might be used, I'm curious if it makes more sense to skip the `config` part, and pass `key` and `block` as two separate arguments.